### PR TITLE
[Reviewer: Graeme] Improve cpp-common coverage

### PR DESCRIPTION
**Pull request by eleanor-merry**
More UTs to improve up cpp-common coverage.

Part of the fix for https://github.com/Metaswitch/cpp-common-test/issues/3

### CODE REVIEW

File: src/ut/baseresolver_test.cpp [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+  _dnsresolver.add_to_cache("cpp-common-test-2.cw-ngv.com", ns_t_a, records);
+
+  AddrInfo ai;
+  ai.transport = IPPROTO_SCTP;
+  ai.port = 3868;
+  DnsRRecord* bl = ResolverUtils::a("cpp-common-test-1.cw-ngv.com", 3600, "3.0.0.1");
+  ai.address = _baseresolver.to_ip46(bl);
+  _baseresolver.blacklist((const AddrInfo)ai);
+
+  //printf("Cache status\n%s", _dnsresolver.display_cache().c_str());
```
Comment (graemerobertson) [inline src/ut/baseresolver_test.cpp]: What's this?

File: src/ut/httpresolver_test.cpp [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /**
- * @file httpresolver_test.cpp UT for HTTPResolver class.
+ * @file httpresolver_test.cpp
```
Comment (graemerobertson) [inline src/ut/httpresolver_test.cpp]: Nice change

File: src/ut/resolver_utils.h [hunk 3]
```diff
@@ -0,0 +1,120 @@
+/**
+ * @file baseresolver_test.cpp UT for BaseResolver class.
```
Comment (graemerobertson) [inline src/ut/resolver_utils.h]: Nope!

File: src/ut/baseresolver_test.cpp [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+  _baseresolver.clear_blacklist();
+}
+
+// Test that blacklisted SRV records aren't chosen
+TEST_F(BaseResolverTest, SRVRecordResolutionWithBlacklist)
+{
+  // Clear the blacklist to start
+  _baseresolver.clear_blacklist();
+
+  // Resolve an A record. Set the port (the transport is always the default)
```
Comment (graemerobertson) [inline src/ut/baseresolver_test.cpp]: Not sure about this comment.

File: src/ut/coverage-not-yet [hunk 5]
```diff
@@ -18,13 +18,8 @@ memcachedstore.cpp
 dnsparser.cpp
 
 # TODO - Incomplete coverage
-logger.cpp
-httpstack.cpp
 diameterstack.cpp
-baseresolver.cpp
 dnscachedresolver.cpp
```
Comment (graemerobertson) [inline src/ut/coverage-not-yet:22]: How are we doing on dnscachedresolver?

File: src/ut/resolver_utils.h [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+  }
+  else
+  {
+    oss << "Unknown (" << ai.transport << ")";
+  }
+  return oss.str();
+}
+
+inline DnsRRecord* a(const std::string& name,
+              int ttl,
```
Comment (graemerobertson) [inline src/ut/resolver_utils.h]: Don't like this indentation, not one bit.

Comment (graemerobertson) [inline src/ut/resolver_utils.h]: Or any of the other functions in this file actually.

**Comment by graemerobertson** [comment]
Ugh, fine.

File: src/ut/coverage-not-yet (hunk 5 above)
Comment (eleanor-merry) [inline src/ut/coverage-not-yet:22]: 48% coverage

**Pull request merged**